inc/field/fieldsfield.class.php

```diff
@@ -0,0 +1,614 @@
+ * ---------------------------------------------------------------------
+ */
+
+namespace GlpiPlugin\Formcreator\Field;
+
+use PluginFieldsContainer;
+use PluginFormcreatorAbstractField;
+use Dropdown;
+use DbUtils;
+use Plugin;
+use Ticket;
+use Change;
+use Problem;
+use Session;
+use Html;
+use OperatingSystem;
+use PluginFieldsDropdown;
+use PluginFieldsField;
+use PluginFormcreatorQuestion;
+use Toolbox;
+use User;
+
+use GlpiPlugin\Formcreator\Exception\ComparisonException;
+use Glpi\Application\View\TemplateRenderer;
+class FieldsField extends PluginFormcreatorAbstractField
+{
+
+   /** @var PluginFieldsField $field */
+   public $field = null;
+
+   /**
+    *
+    * @param array $question PluginFormcreatorQuestion instance
+    */
+   public function __construct(PluginFormcreatorQuestion $question) {
+      $this->question  = $question;
+      if (isset($this->question->fields['values'])) {
+         $decodedValues = json_decode($this->question->fields['values'], JSON_OBJECT_AS_ARRAY);
+         $field_name = $decodedValues['dropdown_fields_field'] ?? '';
+         $fieldObj = new PluginFieldsField();
+         if ($fieldObj->getFromDBByCrit(['name' => $field_name])) {
+            $this->field  = $fieldObj;
+         }
+      }
+   }
+
+   public function isPrerequisites(): bool {
+      return (new Plugin())->isActivated('fields');
+   }
+
+   public static function getFieldsFromBlock($block_id): array {
+
+      global $DB;
+      $optgroup = [];
+
+      if ($block_id == 0) {
+         return $optgroup;
+      }
+
+      // Get all fields for block (except header)
+      $request = [
+         'SELECT' => ['name', 'label'],
+         'FROM'   => PluginFieldsField::getTable(),
+         'WHERE'  => [
+            'AND' => [
+               'plugin_fields_containers_id'      => $block_id,
+               'is_active' => true,
+               ['NOT' => ['type' => 'header']]
+            ]
+         ]
+      ];
+
+      $iterator = $DB->request($request);
+      foreach ($iterator as $row) {
+         $optgroup[$row['name']] = $row['label'];
+      }
+
+      return $optgroup;
+   }
+
+   public function getBlocks(): array {
+      global $DB;
+      $optgroup = [];
+
+      // Get all block for :
+      // Ticket / Change / Problem
+      // Activated
+      // Type -> dom
+      $dbUtils = new DbUtils();
+      $entityRestrict = $dbUtils->getEntitiesRestrictCriteria(PluginFieldsContainer::getTable(), "", "", true, false);
+      if (count($entityRestrict)) {
+         $entityRestrict = [$entityRestrict];
+      }
+
+      $request = [
+         'SELECT' => ['id', 'label'],
+         'FROM'   => PluginFieldsContainer::getTable(),
+         'WHERE'  => [
+            'AND' => [
+               'OR' => [
+                  ['itemtypes' => ['LIKE', '%\"'.Ticket::getType().'\"%']],
+                  ['itemtypes' => ['LIKE', '%\"'.Change::getType().'\"%']],
+                  ['itemtypes' => ['LIKE', '%\"'.Problem::getType().'\"%']],
```

Comment:
I think it would be more reliable to build the allowed itemtypes conditions from the return value of PluginFormcreatorForm::getTargetTypes()

Some extra itemtypes may be added by plugins. If some itemtypes are not supported by Fields they will be ignored because no block will match them.